tests/test_filereader.py

```diff
@@ -1607,40 +1608,37 @@ class TestDeferredRead:
     works as expected
     """
 
-    # Copy one of test files and use temporarily, then later remove.
-    def setup_method(self):
-        self.testfile_name = ct_name + ".tmp"
-        shutil.copyfile(ct_name, self.testfile_name)
-
-    def teardown_method(self):
-        if os.path.exists(self.testfile_name):
-            os.remove(self.testfile_name)
-
     def test_time_check(self):
         """Deferred read warns if file has been modified"""
-        ds = dcmread(self.testfile_name, defer_size="2 kB")
-        from time import sleep
+        with tempfile.TemporaryDirectory() as tdir:
+            p = Path(tdir) / "foo.dcm"
+            shutil.copy(ct_name, p)
+            ds = dcmread(p, defer_size="2 kB")
 
-        sleep(0.1)
-        with open(self.testfile_name, "r+") as f:
-            f.write("\0")  # "touch" the file
+            sleep(0.1)
+            with p.open("r+") as f:
+                f.write("\0")  # "touch" the file
 
-        msg = r"Deferred read warning -- file modification time has changed"
-        with pytest.warns(UserWarning, match=msg):
-            ds.PixelData
+            msg = r"Deferred read warning -- file modification time has changed"
+            with pytest.warns(UserWarning, match=msg):
+                ds.PixelData
 
     def test_file_exists(self):
         """Deferred read raises error if file no longer exists."""
-        ds = dcmread(self.testfile_name, defer_size=2000)
-        os.remove(self.testfile_name)
+        with tempfile.TemporaryDirectory() as tdir:
+            p = Path(tdir) / "foo.dcm"
+            shutil.copy(ct_name, p)
+            ds = dcmread(p, defer_size=2000)
+            os.remove(p)
```

Comment:
Doesn't really matter, but I would use `p.unlink()` instead of mixing `os` and `pathlib`.